antarest/study/web/raw_studies_blueprint.py

```diff
@@ -59,26 +104,42 @@ def get_study(
         output = study_service.get(uuid, path, depth, formatted, parameters)
 
         if isinstance(output, bytes):
-            try:
-                # try to decode string
-                output = output.decode("utf-8")
-            except (AttributeError, UnicodeDecodeError):
-                pass
-
-        json_response = json.dumps(
-            output,
-            ensure_ascii=False,
-            allow_nan=True,
-            indent=None,
-            separators=(",", ":"),
-        ).encode("utf-8")
-        return Response(content=json_response, media_type="application/json")
+            resource_path = pathlib.Path(path)
+            suffix = resource_path.suffix.lower()
+            content_type, encoding = CONTENT_TYPES.get(suffix, (None, None))
+            if content_type == "application/json":
+                # Use `JSONResponse` to ensure to return a valid JSON response
+                # that checks `NaN` and `Infinity` values.
+                output = json.loads(output)
```

Comment:
In case of error, we get a `ValueError`, for instance:

```python
>>> import json
>>> import math
>>> data = {"foo": math.nan}
>>> json.dumps(data, allow_nan=False)
Traceback (most recent call last):
  ...
ValueError: Out of range float values are not JSON compliant
```

This exception will raise a HTTP 500 error. But you're right, it might be better to raise a 422 error to indicate that the configuration data is invalid.

Reply:
I let you do that and i will approve + merge the PR when done &#x1F642;